Army/ArmyLRFull.cs

```diff
@@ -472,106 +399,50 @@ public void ArmyLTs(int quant)
-            hasDarkCaster = true;
-        else
-        {
-            List<InventoryItem> InventoryData = Bot.Inventory.Items;
-            foreach (InventoryItem Item in InventoryData)
-            {
-                if (Item.Name.Contains("Dark Caster") && Item.Category == ItemCategory.Class)
-                {
-                    hasDarkCaster = true;
-                    break;
-                }
-            }
-            if (!hasDarkCaster)
-            {
-                List<InventoryItem> BankData = Bot.Bank.Items;
-                foreach (InventoryItem Item in BankData)
-                {
-                    if (Item.Name.Contains("Dark Caster") && Item.Category == ItemCategory.Class)
-                    {
-                        hasDarkCaster = true;
-                        Core.Unbank(Item.Name);
-                        break;
-                    }
-                }
-            }
-        }
-        if (!hasDarkCaster)
-            ILDC.GetILDC(false);
+        if (item == null)
+            return;
 
-        Core.AddDrop("Legion Token");
-        Core.AddDrop(LRMaterials);
-        Core.AddDrop(LF1);
+        Core.AddDrop(item);
 
-        Core.Logger($"Farming {quant} Revenant's Spellscroll");
-        Bot.Quests.UpdateQuest(2060);
-        Core.RegisterQuests(6897);
-        int i = 1;
-        while (!Bot.ShouldExit && !Core.CheckInventory("Revenant's Spellscroll", quant))
-        {
-            Core.EquipClass(ClassType.Solo);
-            Adv.BestGear(GearBoost.Undead);
-            /*Sells non-full stacks to keep in sync for each LF1 quest item*/
-            if (!Core.CheckInventory("Aeacus Empowered", 50))
-                Core.SellItem("Aeacus Empowered", 0, true);
-            ArmyHunt("judgement", new[] {"Ultra Aeacus"}, "Aeacus Empowered", false, 50);
+        Core.EquipClass(classType);
+        Core.FarmingLogger(item, quant);
 
-            Core.EquipClass(ClassType.Farm);
-            Adv.BestGear(GearBoost.dmgAll);
-            
-            if (!Core.CheckInventory("Tethered Soul", 300))
-                Core.SellItem("Tethered Soul", 0 , true);
-            ArmyHunt("revenant",  new[] {"Forgotten Soul"}, "Tethered Soul", false, 300);
-            
-            if (!Core.CheckInventory("Darkened Essence", 500))
-                Core.SellItem("Darkened Essence", 0, true);
-            ArmyHunt("shadowrealmpast", new[] { "Pure Shadowscythe, Shadow Guardian, Shadow Warrior" }, "Darkened Essence", false, 500);
+        Core.Join(map);
+        WaitCheck();
 
-            Bot.Quests.UpdateQuest(2061);
-            Adv.BestGear(GearBoost.Undead);
-            
-            if (!Core.CheckInventory("Dracolich Contract", 1000))
-                Core.SellItem("Dracolich Contract", 0, true);
-            ArmyHunt("necrodungeon", new[] { "5 Headed Dracolich" }, "Dracolich Contract", false, 1000);
+        Army.SmartAggroMonStart(map, monsters);
 
-            Bot.Drops.Pickup("Revenant's Spellscroll");
```

Comment:
just do a Core.AddDrop(if it isnt in one already,) and a Bot.Wait.ForPickup("item"); at the end